### PR TITLE
Clean up BitmapInvertedIndexReader

### PR DESCRIPTION
**Pull request by jackjlli**
This PR cleans up BitmapInvertedIndexReader

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java [hunk 1]
```diff
@@ -95,32 +90,32 @@ private synchronized ImmutableRoaringBitmap buildRoaringBitmapForIndex(final int
     final int bufferLength = nextOffset - currentOffset;
 
     // Slice the buffer appropriately for Roaring Bitmap
-    ByteBuffer bb = buffer.toDirectByteBuffer(currentOffset, bufferLength);
+    ByteBuffer bb = _buffer.toDirectByteBuffer(currentOffset, bufferLength);
     ImmutableRoaringBitmap immutableRoaringBitmap = null;
     try {
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java]: Recommend removing this try-catch. Returning null will make it even harder to debug

File: pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java [hunk 2]
```diff
@@ -95,32 +90,32 @@ private synchronized ImmutableRoaringBitmap buildRoaringBitmapForIndex(final int
...
+    return _buffer.getInt(index * Integer.BYTES);
   }
 
   private void load(PinotDataBuffer indexDataBuffer)
       throws IOException {
-    final int lastOffset = indexDataBuffer.getInt(numberOfBitmaps * Integer.BYTES);
+    final int lastOffset = indexDataBuffer.getInt(_numberOfBitmaps * Integer.BYTES);
     assert lastOffset == indexDataBuffer.size();
-    this.buffer = indexDataBuffer;
+    this._buffer = indexDataBuffer;
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java]: Remove `this.`

File: pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java [hunk 3]
```diff
@@ -47,8 +43,7 @@
    */
   public BitmapInvertedIndexReader(PinotDataBuffer indexDataBuffer, int cardinality)
       throws IOException {
-    this.file = file;
-    numberOfBitmaps = cardinality;
+    _numberOfBitmaps = cardinality;
     load(indexDataBuffer);
```
[the remark this answers is not in the document]
Comment (jackjlli) [reply pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java]: Done.

File: pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java (hunk 1 above)
Comment (jackjlli) [reply pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java]: Done.

File: pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java (hunk 2 above)
Comment (jackjlli) [reply pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java]: Done.

File: pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java [hunk 4]
```diff
@@ -31,12 +30,9 @@
 public class BitmapInvertedIndexReader implements InvertedIndexReader<ImmutableRoaringBitmap> {
   public static final Logger LOGGER = LoggerFactory.getLogger(BitmapInvertedIndexReader.class);
 
-  final private int numberOfBitmaps;
-  private volatile SoftReference<SoftReference<ImmutableRoaringBitmap>[]> bitmaps = null;
-
-  private PinotDataBuffer buffer;
-
-  private File file;
+  private final int _numberOfBitmaps;
```
[the remark this answers is not in the document]
Comment (jackjlli) [reply pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java]: Done.

File: pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java [hunk 5]
```diff
@@ -31,25 +30,22 @@
...
-  public BitmapInvertedIndexReader(PinotDataBuffer indexDataBuffer, int cardinality)
-      throws IOException {
-    this.file = file;
-    numberOfBitmaps = cardinality;
-    load(indexDataBuffer);
+  public BitmapInvertedIndexReader(PinotDataBuffer indexDataBuffer, int cardinality) {
+    _numBitmaps = cardinality;
+    _buffer = indexDataBuffer;
+
+    final int lastOffset = _buffer.getInt(_numBitmaps * Integer.BYTES);
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java]: Make this a precondition check

File: pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java [hunk 6]
```diff
@@ -31,25 +30,22 @@
...
 
-  final private int numberOfBitmaps;
-  private volatile SoftReference<SoftReference<ImmutableRoaringBitmap>[]> bitmaps = null;
-
-  private PinotDataBuffer buffer;
-
-  private File file;
+  private final int _numBitmaps;
+  private volatile SoftReference<SoftReference<ImmutableRoaringBitmap>[]> _bitmaps = null;
+  private final PinotDataBuffer _buffer;
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java]: (nit) move this before `_bitmaps`

File: pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java (hunk 5 above)
Comment (jackjlli) [reply pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/BitmapInvertedIndexReader.java]: Updated.

**Pull request merged**